Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs

```diff
@@ -5402,6 +5386,46 @@ protected override void UpdateTractiveForce(float elapsedClockSeconds, float t,
         }
 
 
+        /// <summary>
+        /// Normalise crank angle so that it is a value between 0 and 360 starting at the real crank angle difference
+        /// </summary>
+        private float NormalisedCrankAngle( int cylinderNumber)
+        {
+            float realCrankAngleRad = (float)(LocomotiveAxles[0].AxlePositionRad);
+            float normalisedCrankAngleRad = 0;
+
+            realCrankAngleRad = (float)(MathHelper.WrapAngle(realCrankAngleRad));
+
+            if (realCrankAngleRad < 0)
+            {
+                realCrankAngleRad = (float)(2.0f * Math.PI + realCrankAngleRad); // angle must be maintained in a +ve range, ie 0 - 360
+            }
+
+            
+            if (cylinderNumber == 0) // initial cylinder
```

Comment:
I don't see why the first cylinder should be different to the rest. Wouldn't be better to set WheelCrankAngleDiffRad[0]=0 ?

Reply:
It was already set = 0, so I have removed this irrelevant bit of code.